File: go-algorithm/leetcode/problem70.go
package leetcode

// 70. Climbing Stairs
// https://leetcode.com/problems/climbing-stairs/
func climbStairs(n int) int {
	if n == 1 {
		return 1
	}
	dp := []int{1, 2}
	for i := 2; i < n; i++ {
		dp[1], dp[0] = dp[0]+dp[1], dp[1]
	}
	return dp[1]
}

File: java-algo/src/main/java/com/leetcode/Problem70.java
package com.leetcode;

/**
 * 70. Climbing Stairs
 * https://leetcode.com/problems/climbing-stairs/
 */
public class Problem70 {
    public int climbStairs(int n) {
        int a, b = 0, c = 1;
        for (int i = 1; i <= n; i++) {
            a = b;
            b = c;
            c = a + b;
        }
        return c;
    }
}
